lib/common_test/src/ct_telnet.erl

```diff
@@ -502,6 +580,21 @@ expect(Connection, Patterns) -> term()
 
 Equivalent to [`ct_telnet:expect(Connections, Patterns, [])`](`expect/3`).
 """.
+-spec expect(Connection, Patterns) -> {'ok', Match} | {'ok', Match, HaltReason}
+              | {'error', Reason} | no_return()
+              when Connection :: connection(),
+                   Patterns :: Pattern | [Pattern],
+                   Pattern :: PatternString | {Tag, PatternString},
+                   Tag :: term(),
+                   PatternString :: unicode:charlist(),
+                   Match :: MatchResult | [MatchResult],
+                   MatchResult :: [CaptureData | {Tag, CaptureData}]
+                                | [[CaptureData | {Tag, CaptureData}]],
+                   CaptureData :: string()
```

Comment:
I think this should be fixed in code, so both {'error', string(), binary()} and {'incomplete', string(), binary()} are not possible.

Reply:
I think it sounds like an improvement. But this is a not backwards compatible change so it is to late to fix it for OTP-27. 
You could decide on the return value that you think makes sense and write something about it in the documentation to 
sort of introduce that it may be changed for OTP-28.

Reply:
Maybe if we think it worth it we could use the approach of adding a new better API function and deprecating this one in next release to avoid backwards incompatibility.

Reply:
Fixed in documentation.